WordPress/wordpress-playground
# [Exploration] Add Soap PHP library support

**This PR is an experiment and it's not intended to be merged.**

This PR adds Soap support to PHP, by building PHP with `--enable-soap`.

## Progress on adding Soap support

I was able to build PHP 8.0 with Soap and can see it's enabled in PHP Info.

To recompile PHP you can use `npm run recompile:php:web:asyncify:8.0` for a single version, or all versions using `npm run recompile:php`.
See full list of[ recompile commands here.](https://github.com/WordPress/wordpress-playground/blob/1e512482faa926f28d6d62333d12d491c22b05be/package.json#L25)

### TODO: Check if Soap works in Playground

We need to check if Soap works in Playground. 

I tried a [test script from Stack overflow](https://stackoverflow.com/a/17067516) but it failed with this error

```
Fatal error: Uncaught SoapFault exception: [WSDL] SOAP-ERROR: Parsing WSDL: Couldn't load from 'https://www.w3schools.com/xml/tempconvert.asmx?WSDL' : failed to load external entity "https://www.w3schools.com/xml/tempconvert.asmx?WSDL" in /wordpress/soap-test.php:3 Stack trace: #0 /wordpress/soap-test.php(3): SoapClient->__construct('https://www.w3s...') #1 {main} thrown in /wordpress/soap-test.php on line 3
```

Fixes #1236

## Testing Instructions (or ideally a Blueprint)

- Open Playground locally with this URL http://127.0.0.1:5400/website-server/?php=8.0&url=%2Fphpinfo.php
- Confirm that the Soap server and client are enabled
- TODO Confirm that Soap works
